classes/i18n/LocaleConversion.php

```diff
@@ -24,18 +24,41 @@
 namespace PKP\i18n;
 
 use Illuminate\Support\Arr;
+use PKP\core\Core;
 use PKP\facades\Locale;
 
 class LocaleConversion
 {
+    /**
+     * Get ISO639-2b array
+     */
+    protected static function getISO6392b(): array
+    {
+        $iso6392BFile = Core::getBaseDir() . '/' . PKP_LIB_PATH . '/lib/vendor/sokil/php-isocodes-db-i18n/databases/iso_639-2.json';
+        if (!file_exists($iso6392BFile)) {
+            return [];
+        }
+        return json_decode(file_get_contents($iso6392BFile), true);
+    }
+
     /**
      * Translate the ISO 2-letter language string (ISO639-1) into a ISO compatible 3-letter string (ISO639-2b).
      */
     public static function get3LetterFrom2LetterIsoLanguage(?string $iso2Letter): ?string
     {
         assert(strlen($iso2Letter) === 2);
-        $locale = Arr::first(Locale::getLocales(), fn (LocaleMetadata $locale) => $locale->getIsoAlpha2() === $iso2Letter);
-        return $locale ? $locale->getIsoAlpha3() : null;
+        $iso6392b = null;
+        $languages = self::getISO6392b();
+        foreach (reset($languages) as $languageRaw) {
+            if (array_key_exists('alpha_2', $languageRaw) && $languageRaw['alpha_2'] === $iso2Letter) {
```

Comment:
Not sure if it's ok for your use-case, but I'd use: `($languageRaw['alpha_2'] ?? null) === $iso2Letter` and below `$iso6392b = $languageRaw['bibliographic'] ?? $languageRaw['alpha_3']`